src/openms/include/OpenMS/ANALYSIS/ID/FIAMSDataProcessor.h

```diff
@@ -0,0 +1,209 @@
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+// 
+// --------------------------------------------------------------------------
+// $Maintainer: Chris Bielow $
+// $Authors: Marc Sturm, Hendrik Weisser, Chris Bielow $
+// --------------------------------------------------------------------------
+ 
+#pragma once
+
+#include <OpenMS/CONCEPT/ClassTest.h>
+#include <OpenMS/FORMAT/FeatureXMLFile.h>
+#include <OpenMS/TRANSFORMATIONS/FEATUREFINDER/PeakWidthEstimator.h>
+#include <OpenMS/TRANSFORMATIONS/RAW2PEAK/PeakPickerCWT.h>
+#include <OpenMS/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.h>
+#include <OpenMS/ANALYSIS/ID/AccurateMassSearchEngine.h>
+#include <OpenMS/FORMAT/MzTabFile.h>
+#include <OpenMS/FORMAT/MzTab.h>
+#include <OpenMS/FILTERING/NOISEESTIMATION/SignalToNoiseEstimatorMedianRapid.h>
+#include <OpenMS/FILTERING/BASELINE/MorphologicalFilter.h>
+#include <OpenMS/KERNEL/SpectrumHelper.h>
+#include <OpenMS/ANALYSIS/OPENSWATH/SpectrumAddition.h>
+#include <OpenMS/FILTERING/SMOOTHING/SavitzkyGolayFilter.h>
+#include <OpenMS/FORMAT/MzMLFile.h>
+
+using namespace std;
+
+namespace OpenMS
+{
+  /**
+    description
+  */
+  class OPENMS_DLLAPI FIAMSDataProcessor 
+  {
+public:
+    /// Default constructor
+    FIAMSDataProcessor(
+      String filename, 
+      String dir_input, 
+      String dir_output, 
+      float resolution, 
+      String polarity, 
+      String db_mapping, 
+      String db_struct, 
+      String positive_adducts, 
+      String negative_adducts, 
+      bool store_progress=true,
+      float min_mz=50, 
+      float max_mz=1500, 
+      float bin_step=20
+    );
+
+    /// Default desctructor
+    ~FIAMSDataProcessor();
+
+    /// Copy constructor
+    FIAMSDataProcessor(const FIAMSDataProcessor& cp);
```

Comment:
```suggestion
    FIAMSDataProcessor(const FIAMSDataProcessor& cp) = default;
```

saves some space in the cpp file